conformance/utils/http/http.go

```diff
@@ -38,11 +38,13 @@ type ExpectedResponse struct {
 	// expected to match Request.
 	ExpectedRequest *ExpectedRequest
 
-	// TODO: move these into a dedicated type named Response
-	// ref: https://github.com/kubernetes-sigs/gateway-api/issues/1384
-	StatusCode    int
-	Headers       map[string]string
-	AbsentHeaders []string
+	// BackendSetResponseHeaders is a set of headers
+	// the echoserver should set in its response.
+	BackendSetResponseHeaders map[string]string
```

Comment:
an alternative is to just have the tests specify the `X-Echo-Set-Header` header themselves, but this seemed like a better solution so people adding tests don't have to know how that exactly works